redhat-developer/developers.redhat.com
# DO NOT MERGE [DEVELOPER-5432] Assembly Code Snippets

This implements a new code snippet assembly type and uses Highlight.js
for code syntax highlighting. Some remaining tasks include determining a
syntax highlighting theme that is on-brand and attractive.

### JIRA Issue Link
* https://issues.jboss.org/browse/DEVELOPER-5432

### Verification Process

Go here:

Relative URL: `/node/add/article`

Do this:

* Enter any content for required fields
* Once you reach the Content field, you should notice that you now have an option of using a Code Snippet assembly type.
* Add a new code snippet assembly type
* Enter some name for the code snippet that makes sense to you
* In the Code, long text unformatted field, enter some code like:

```
var msg = 'hello world';
console.log(msg);
```

* Finish filling out any required fields and save the edit form to create the Article

You should see this:

* You should be redirected to your newly created Article containing a code snippet assembly

![githubhighlightjstheme](https://user-images.githubusercontent.com/7155034/46836562-e2c94680-cd66-11e8-9220-05f57dfa38a8.png)
